### PR TITLE
app/vmauth: dump requests that failed the route rules to stderr

### PR DESCRIPTION
**Pull request by hagen1778**
Additional info from the dump can be used to debug rotuing rules.

https://pkg.go.dev/net/http/httputil#DumpRequest

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Comment by AndrewChubatiuk** [comment]
can we dump only vmauth related data like Host header and paths? as I have doubts regarding logs readability in case of 10+ headers

**Comment by AndrewChubatiuk** [comment]
can we display only data which is related to vmauth functionality as Host header, path, etc?

File: app/vmauth/main.go [hunk 1]
```diff
@@ -198,7 +199,12 @@ func processRequest(w http.ResponseWriter, r *http.Request, ui *UserInfo) {
 				return
 			}
 			missingRouteRequests.Inc()
-			httpserver.Errorf(w, r, "missing route for %s", u.String())
+			dump, err := httputil.DumpRequest(r, false)
+			if err != nil {
+				httpserver.Errorf(w, r, "failed to dump HTTP request: %s", err)
+				return
+			}
+			httpserver.Errorf(w, r, "missing route for %q (%q)", u.String(), dump)
```
Comment (f41gh7) [inline app/vmauth/main.go:207]: Could it expose authorization header values and other sensitive headers information to the logs?

Comment (hagen1778) [reply app/vmauth/main.go:207]: Yes. We can make dumping optional, for debug purposes.

**Comment by hagen1778** [comment]
> can we return only data which is related to vmauth functionality as Host header, path, etc?

Yes, we can do that as well.

File: app/vmauth/main.go (hunk 1 above)
Comment (hagen1778) [reply app/vmauth/main.go]: Updated PR

**Review by f41gh7 — APPROVED** [review]
LGTM

**Review by AndrewChubatiuk — APPROVED** [review]

**Pull request merged**